### DISCUSSION TITLE
MorphToMany pivots do not have the morph type column when detached

### DISCUSSION THREAD

**Discussion by gerardnll** [laravel/framework#43760] [General]
If you have a MorphPivot entity with a custom class, when attaching a record, you get the pivot with all the fields loaded on the model, but if you detach it, you only get the IDs of both sides, so in case of morphPivots, you do not get the type/class of the entity so there's information missing and you cannot obtain the morph relation from the pivot because it throws an error.

This can be fixed by referring to the pivotParent attribute on the MorphPivot but it's kind of weird as in the attach process you can refer to the relation. My solution is using the pivotParent attribute on both attach and detach events so we behave the same way in both situations.

**Comment by Sophist-UK** [comment]
In principal at last, this actually sounds like a genuine bug - is it worth creating a Github issue for it?

**Reply by gazben** [reply]
I've run into this issue and I too think this is a bug. I think it worth an issue.
The `pivotParent` attribute does not always results in the desired object.

**Comment by gazben** [comment]
I've suggested a possible solution here: https://github.com/laravel/framework/pull/48432
